assets/javascripts/admin_assets.js

```diff
@@ -117,11 +120,15 @@ function deleteAsset(assetId) {
     $.ajax({
         url: '/api/v1/assets/' + assetId,
         method: 'DELETE',
+	dataType: 'json',
         success: function() {
-            $('#asset_' + assetId).remove();
+	    var table = $("#assets").DataTable();
+	    table.row($("#asset_" + assetId)).remove().draw();
```

Comment:
Maybe we shouldn't even remove the row here. It would be only a temporary removal which does not persist reloading the page. That's why you added the flash message and trying to fake the view update contradicts that message.